libdrgn/helpers.h

```diff
@@ -43,4 +45,102 @@ struct drgn_error *linux_helper_find_task(struct drgn_object *res,
 					  const struct drgn_object *ns,
 					  uint64_t pid);
 
+/**
+ * Iterator convention:
+ *
+ * For all of the iterators defined below, the convention for each of the
+ * `*_next` functions is that upon returning, `*ret` will point to space
+ * allocated inside of `iter`. The caller is free to do what they wish with
+ * this return value, but should note that it will be overwritten the next time
+ * the `*_next` function is called.
+ */
+
+struct linux_helper_radix_tree_iter_frame;
+
```

Comment:
You don't need this forward declaration. `typedef struct foo bar;` provides an implicit forward declaration of `struct foo`, and `DEFINE_VECTOR_TYPE` effectively contains that.